src/HDF5.jl

```diff
@@ -942,7 +942,7 @@ name(attr::Attribute) = h5a_get_name(attr)
 function Base.keys(x::Union{Group,File})
     checkvalid(x)
     children = sizehint!(String[], length(x))
-    h5l_iterate(x, H5_INDEX_NAME, H5_ITER_INC) do group, name, info, data
```

Comment:
The `data` argument is simply wrong here since `data` actually refers to the deafult argument `idx=0`  ?

Reply:
`data` used to always be `C_NULL`, since that's what we handed as the user data argument to the low-level `h5l_iterate` call. In what this PR does now, it just doesn't exist anymore since the function wrapper is utilizing and consuming the user data pointer.